### PR TITLE
Explore: move items out of topnav & use canvas variant for toolbar buttons

### LINKED ISSUE #60974: Explore: move items out of topnav & use canvas variant for toolbar buttons
From a UX feedback session it emerged that the current state of Explore navigation is suboptimal: Toolbar buttons get moved out of topnav into a toolbar in each pane when in split mode, moreover opening and closing a pane causes them to be placed back in the topnav.

Image

Also, when topnav is enabled and in split mode, the buttons in each pane have a transparent background, making it hard for users to visually identify them.

Image

The suggested solution was to not use the topnav for any button except for the "copy shortened link" icon button, which is consistent with dashboards, but instead always render the toolbar in each pane (even when not in split mode).
Additionally each button in the toolbar should use the `canvas` variant, which is the intended use of toolbar buttons living outside of the topnav.

### PR DESCRIPTION
**Pull request by Elfo404**
**What is this feature?**

Screenshot 2022-12-21 at 14 47 33

As discussed in a UX feedback session, having buttons changing their rendering location when changing the split state is less than ideal from a UX perspective. this reintroduces the explore toolbar also when non in split mode - moving back all but the share button there - and fixes a few minor issues:

- bfa7292cdda150d1f8a9ce02b8122371b237aa1f avoids rendering an extra element in the toolbar which resulted in extra spacing when using datasource with live support: 
Before:
Screenshot 2022-12-21 at 14 50 08

After:
Screenshot 2022-12-21 at 14 45 11

- c1001b514f3e6184e1d41cdd506caa7cfae1de08 fixes `leftItems` spacing when no title or page icon is provided which caused the ds picker to be misaligned with page content (cc. @grafana/user-essentials):
Before:
Screenshot 2022-12-21 at 14 49 04

After:
Screenshot 2022-12-21 at 14 46 55

- 9d34cc7416de750665a6f3f9cb233722447ce8e6 avoids disabling the split resize button when live tailing. i think this was unintentional in the first place

Fixes #60974

### CODE REVIEW

**Comment by radiorental** [comment]
Reviewed with @Elfo404, looks great from a UX perspective, do not foresee any downsides to this enhancement.

**Comment by ifrost** [comment]
The data source selector looks a bit confusing to me 🤔 it has no label, it's just pure dropdown. Maybe we should at least add a label like in dashboards?

Explore:

Screenshot 2023-01-04 at 12 24 36

Dashboards:

Screenshot 2023-01-04 at 12 24 29

**Comment by Elfo404** [comment]
@ifrost given the ds picker in explore was always like this do you think this needs to be addressed in this PR (which is basically about reverting the toolbar behaviour to before the topnav was introduced)?

**Comment by ifrost** [comment]
> @ifrost given the ds picker in explore was always like this do you think this needs to be addressed in this PR (which is basically about reverting the toolbar behaviour to before the topnav was introduced)?

It's a bit different now, before in non-split mode the dropdown was next to Explore title, now it looks a bit lonely/floating/out-of-context. If Pauric is okay with it I don't think it needs to be addressed at all.

**Review by gelicia — APPROVED** [review]

File: packages/grafana-ui/src/components/PageLayout/PageToolbar.tsx [hunk 1]
```diff
@@ -190,7 +190,7 @@ const getStyles = (theme: GrafanaTheme2) => {
       min-width: 0;
     `,
     h1Styles: css`
-      margin: 0;
+      margin: 0 ${spacing(1)} 0 0;
```
Comment (torkelo) [inline packages/grafana-ui/src/components/PageLayout/PageToolbar.tsx:193]:
```suggestion
      margin: ${spacing(0, 1, 0, 0)};
```

**Review by torkelo — COMMENTED** [review]
Looks good!

**Pull request merged**